### ISSUE TITLE
Mint.HTTPError "the given data exceeds the request window size" when sending a large request

### ISSUE THREAD

**Issue by knewter** [thmsmlr/instructor_ex#31]
When sending a larger document for summary in my instructor pipeline, I run into the following error:

```
[error] Task #PID<0.36499.0> started from #PID<0.36490.0> terminating
[info]** (Mint.HTTPError) the given data exceeds the request window size, which is 65535. The server will refill the window size of the request when ready. This will be handled transparently by stream/2.
[info]    (req 0.4.11) lib/req.ex:978: Req.request!/2
[info]    (instructor 0.0.5) lib/instructor/adapters/openai.ex:80: Instructor.Adapters.OpenAI.do_chat_completion/2
[info]    (instructor 0.0.5) lib/instructor.ex:430: Instructor.do_chat_completion/3
[info]    (summarizer 0.1.0) lib/summarizer.ex:20: Summarizer.summarize/1
[info]    (summarizer 0.1.0) lib/summarizer_web/live/link_live/show.ex:27: anonymous fn/1 in SummarizerWeb.LinkLive.Show.handle_info/2
[info]    (elixir 1.16.1) lib/task/supervised.ex:101: Task.Supervised.invoke_mfa/2
[info]    (elixir 1.16.1) lib/task/supervised.ex:36: Task.Supervised.reply/4
```

There's a similar error reported for cory o'daniel's k8s client, also using mint for its client: https://github.com/coryodaniel/k8s/issues/220

I'm not sure what to do. For now, I just accept that some documents I cannot handle, but I'd prefer it if I could get around this. I'm fine spelunking and figuring it out but I thought I'd file an issue first in case I've missed something obvious.

Thanks for instructor_ex, it's great!

**Comment by TwistingTwists** [comment]
https://github.com/thmsmlr/instructor_ex/pull/26/files

See the above example. 

You can pass on some options to mint

**Comment by knewter** [comment]
I tried this:

```
config :instructor,
  adapter: Instructor.Adapters.OpenAI,
  openai: [
    api_key: System.get_env("OPENAI_API_KEY"),
    http_options: [
      receive_timeout: 240_000,
      connect_options: [
        client_settings: [
          initial_window_size: 2_000_000
        ]
      ]
    ]
  ]
```

this is correct, because if i give another option in there mint will blow up with an unknown option error. however, this doesn't seem to bump the window up successfully - the error message is identical.

tl;dr still don't know how to send large documents over and it's not from lack of digging in and trying. it's possible that using streaming mode mitigates this, i don't actually know

**Comment by TwistingTwists** [comment]
can you please try with these options once? 

```
http_options: [receive_timeout: 60_000, connect_options: [protocols: [:http2]]]
```

**Comment by thmsmlr** [comment]
Hmm, very weird. When y'all figure out a good config, lets merge it in as the default config

**Comment by knewter** [comment]
@TwistingTwists this fixed it, thank you

**Comment by knewter** [comment]
@thmsmlr i closed this but given we found a good config it seemed worth pinging you so it didn't get lost in the shuffle
